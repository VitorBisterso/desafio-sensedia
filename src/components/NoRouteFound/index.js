import React from 'react';
import { useLocation } from 'react-router-dom';

import './styles.css';

function NoRouteFound() {
  const { pathname } = useLocation();

  return (
    <div className="noRouteFound-container">
      <h2>
        Não há nenhuma rota com o caminho de <code>{pathname}</code>
      </h2>
    </div>
  );
}

export default NoRouteFound;
