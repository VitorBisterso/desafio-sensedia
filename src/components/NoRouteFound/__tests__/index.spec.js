import React from 'react';
import renderer from 'react-test-renderer';

import NoRouteFound from '../index';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/anything',
  }),
}));

it('renders correctly', () => {
  const tree = renderer.create(<NoRouteFound />).toJSON();
  expect(tree).toMatchSnapshot();
});
