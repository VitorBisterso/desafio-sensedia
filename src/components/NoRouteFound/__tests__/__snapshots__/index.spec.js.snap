// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="noRouteFound-container"
>
  <h2>
    Não há nenhuma rota com o caminho de 
    <code>
      localhost:3000/anything
    </code>
  </h2>
</div>
`;
